### DISCUSSION TITLE
Why self updating Typst binary is disabled if compiled from source (`cargo install`)?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [typst/typst#2152] [Q&A] [answered]
I've compiled the Typst from the source, but the help message shows `update   Self update the Typst CLI (disabled)` and when I run `typst update` I get:

```
error: self-updating is not enabled for this executable, please update with the package manager or mechanism used for initial installation.
```

Why it is disabled and how to compile the same binary as in the release, which can actually self-update?

**Comment by PgBiel** [comment]
Self-update is opt-in, you need to enable the corresponding feature. Something like

```
cargo install --git https://github.com/typst/typst.git -F "self-update"
```

(or similar command) should work.

**Reply by Andrew15-5** [reply]
Thanks. This wasn't in the release notes, so it's really confusing that this is an opt-in feature.

**Reply by Andrew15-5** [reply]
Do you know why is it an opt-in feature?

**Reply by laurmaedje** [reply, marked as answer]
Because it's problematic for package manager installations. And when you're using `cargo install`, you already have an update mechanism built into cargo. So it's mostly useful for people that download the pre-built binaries from GitHub and for those, it is enabled.

**Reply by Andrew15-5** [reply]
Even though `cargo install` works, I don't want to wait for another compilation of a new version. I don't want to download a new version from the release page either, because I can use the self-update feature.

**Reply by pepa65** [reply]
I downloaded for the last time from the Release page, so in the future I can just do `typst update`!

**Comment by pepa65** [comment]
> cargo install --git https://github.com/typst/typst.git -F "self-update"

That would be great to include in the readme or the wiki!
